test/AutoRest.TestServer.Tests/lro.cs

```diff
@@ -0,0 +1,1455 @@
+        });
+
+        [Test]
+        public Task LROPutNonResourceAsyncInRetry_Sync() => Test((host, pipeline) =>
+        {
+            var value = new Sku();
+            var operation = new LrOSOperations(ClientDiagnostics, pipeline, host).StartPutAsyncNonResourceOperation(value);
+            var result = operation.WaitForCompletion();
+            Assert.AreEqual("100", result.Value.Id);
+            Assert.AreEqual("sku", result.Value.Name);
+        });
+
+        [Test]
+        public Task LROPutNonResourceInRetry() => Test(async (host, pipeline) =>
+        {
+            var value = new Sku();
+            var operation = await new LrOSOperations(ClientDiagnostics, pipeline, host).StartPutNonResourceOperationAsync(value);
+            var result = await operation.WaitForCompletionAsync().ConfigureAwait(false);
+            Assert.AreEqual("100", result.Value.Id);
+            Assert.AreEqual("sku", result.Value.Name);
+        });
+
+        [Test]
+        public Task LROPutNonResourceInRetry_Sync() => Test((host, pipeline) =>
+        {
+            var value = new Sku();
+            var operation = new LrOSOperations(ClientDiagnostics, pipeline, host).StartPutNonResourceOperation(value);
+            var result = operation.WaitForCompletion();
+            Assert.AreEqual("100", result.Value.Id);
+            Assert.AreEqual("sku", result.Value.Name);
+        });
+
+        [Test]
+        public Task LROPutSubResourceAsyncInRetry() => Test(async (host, pipeline) =>
+        {
+            var value = new SubProduct();
+            var operation = await new LrOSOperations(ClientDiagnostics, pipeline, host).StartPutAsyncSubResourceOperationAsync(value);
+            var result = await operation.WaitForCompletionAsync().ConfigureAwait(false);
+            Assert.AreEqual("100", result.Value.Id);
+            Assert.AreEqual("Succeeded", result.Value.ProvisioningState);
+        });
+
+        [Test]
+        public Task LROPutSubResourceAsyncInRetry_Sync() => Test((host, pipeline) =>
+        {
+            var value = new SubProduct();
+            var operation = new LrOSOperations(ClientDiagnostics, pipeline, host).StartPutAsyncSubResourceOperation(value);
+            var result = operation.WaitForCompletion();
+            Assert.AreEqual("100", result.Value.Id);
+            Assert.AreEqual("Succeeded", result.Value.ProvisioningState);
+        });
+
+        [Test]
+        public Task LROPutSubResourceInRetry() => Test(async (host, pipeline) =>
+        {
+            var value = new SubProduct();
+            var operation = await new LrOSOperations(ClientDiagnostics, pipeline, host).StartPutSubResourceOperationAsync(value);
+            var result = await operation.WaitForCompletionAsync().ConfigureAwait(false);
+            Assert.AreEqual("100", result.Value.Id);
+            Assert.AreEqual("Succeeded", result.Value.ProvisioningState);
+        });
+
+        [Test]
+        public Tas
```

Comment:
The full pipeline should be retrying on 500 we might need to update the Azure.Core version.

Reply:
Give me a bit to double check this.

Reply:
The original call for each of these is returning 500, which means the pipeline is not retrying it. If we have to update Azure.Core for it to work, we should do that as a separate PR, and try to enable any tests that link to https://github.com/Azure/autorest.csharp/issues/398 in the test project.